src/nimble_model_data.cc

```diff
@@ -383,19 +383,19 @@ void ModelData::SetReferenceCoordinates(const nimble::GenesisMesh &mesh)
     throw std::logic_error("\n**** Error in ModelData: label 'reference_coordinate' not found.\n");
   }
 
-  auto reference_coordinate = Viewify(GetNodeData(field_id), dim_);
+  auto reference_coordinate = GetNodeData(field_id);
   int num_nodes = static_cast<int>(mesh.GetNumNodes());
   if (dim_ == 2) {
     for (int i=0 ; i<num_nodes ; i++) {
-      reference_coordinate(i, 0) = ref_coord_x[i];
-      reference_coordinate(i, 1) = ref_coord_y[i];
+      reference_coordinate[i * dim_ + 0] = ref_coord_x[i];
```

Comment:
the new code works with flattened vectors, is a v(node,dim) accessor out of the question? I'd prefer not having to recall which way the data is striped/laid out

Reply:
Inside 'nimble::ModelData' or 'nimble_kokkos::ModelData', you can choose to do either: 
- use the raw pointer or Viewify (for 'nimble::ModelData')
- use the Kokkos::View or Viewify (for 'nimble_kokkos::ModelData')

I can fix that. Actually with Viewify, I could merge both implementations of 'SetReferenceCoordinates'

Reply:
Fixed and moved the routine to the base class (as Viewify makes it independent).